### DISCUSSION TITLE
Format of EC points returned by EVP_PKEY_get1_encoded_public_key()

### DISCUSSION THREAD

**Discussion by H5117** [openssl/openssl#22835] [Q&A] [answered]
Documentation for `EVP_PKEY_get1_encoded_public_key()` says nothing about what format of EC points it returns: compressed or uncompressed? My experiments show that it returns uncompressed format. Is it always true, or is there a way to control this behavior?

I tried to call `EVP_PKEY_set_utf8_string_param(pkey, OSSL_PKEY_PARAM_EC_POINT_CONVERSION_FORMAT, OSSL_PKEY_EC_POINT_CONVERSION_FORMAT_COMPRESSED)` before `EVP_PKEY_get1_encoded_public_key()`, but it does not affect the result.

**Comment by mattcaswell** [comment, marked as answer]
`EVP_PKEY_get1_encoded_public_key` always returns uncompressed format for EC keys.

Use `EVP_PKEY_get_octet_string_param` with the key `OSSL_PKEY_PARAM_PUB_KEY` after having set the point conversion key format to `OSSL_PKEY_EC_POINT_CONVERSION_FORMAT_COMPRESSED`. 

However, see the description of `OSSL_PKEY_PARAM_PUB_KEY` on the following man page since some older OpenSSL provider versions do not correctly honour this:

https://www.openssl.org/docs/man3.2/man7/EVP_PKEY-EC.html

**Reply by H5117** [reply]
Thank you.

> EVP_PKEY_get1_encoded_public_key always returns uncompressed format for EC keys.

It would be nice to add this to documentation.
